lesson_26/api/java/api_app/src/main/java/com/codedifferently/lesson26/web/MediaItemsController.java

```diff
@@ -29,4 +38,53 @@ public GetMediaItemsResponse getItems() {
     var response = GetMediaItemsResponse.builder().items(responseItems).build();
     return response;
   }
+
+  @GetMapping("/items/{id}")
+  public ResponseEntity<MediaItemResponse> getItemById(@PathVariable("id") String id) {
+    Set<MediaItem> items = library.search(SearchCriteria.builder().id(id).build());
+    if (items.isEmpty()) {
+      return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+    }
+    MediaItem item = items.iterator().next();
+    MediaItemResponse response = MediaItemResponse.from(item);
+    return ResponseEntity.ok(response);
+  }
+
+  @PostMapping("/items")
```

Comment:
My only error is that it expects the status 200 , but gets 400. Fixing this has been an issue 
chatgpt expects me to manually include a book if statement , or a bindingResult <- both still gives the same 400 error.

Reply:
So you had two problems. For one, you were using the wrong `@RequestBody` annotation so the request message was not being read correctly. Secondly, you should have preferred to accept the `CreateMediaItemRequest` type which is the type of request the test anticipates sending.